sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/BridgeInternal.java

```diff
@@ -799,4 +801,12 @@ public static void  initializeAllAccessors() {
         DirectConnectionConfig.initialize();
         CosmosAsyncClient.initialize();
     }
+
+    @Warning(value = INTERNAL_USE_ONLY_WARNING)
+    public static boolean isPartialPartitionKeyQuery(DocumentCollection collection, PartitionKey partitionKey) {
```

Comment:
Use ImplementationBridgeHelper pattern instead? - Also shouldn't we validate that partitionKey is of PartitionKind.MULTI-HASH as well?

Reply:
do you mean I should move this code bit into the ImplementationBridgeHelper? 

we also do check that the pkdef is multihash in the lines right below